bundles/org.openhab.binding.gree/src/main/java/org/openhab/binding/gree/internal/handler/GreeAirDevice.java

```diff
@@ -0,0 +1,721 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.gree.internal.handler;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.gree.internal.GreeCryptoUtil;
+import org.openhab.binding.gree.internal.GreeException;
+import org.openhab.binding.gree.internal.gson.GreeBindRequestDTO;
+import org.openhab.binding.gree.internal.gson.GreeBindRequestPackDTO;
+import org.openhab.binding.gree.internal.gson.GreeBindResponseDTO;
+import org.openhab.binding.gree.internal.gson.GreeBindResponsePackDTO;
+import org.openhab.binding.gree.internal.gson.GreeExecCommandDTO;
+import org.openhab.binding.gree.internal.gson.GreeExecResponseDTO;
+import org.openhab.binding.gree.internal.gson.GreeExecResponsePackDTO;
+import org.openhab.binding.gree.internal.gson.GreeExecuteCommandPackDTO;
+import org.openhab.binding.gree.internal.gson.GreeReqStatusDTO;
+import org.openhab.binding.gree.internal.gson.GreeReqStatusPackDTO;
+import org.openhab.binding.gree.internal.gson.GreeScanResponseDTO;
+import org.openhab.binding.gree.internal.gson.GreeStatusResponseDTO;
+import org.openhab.binding.gree.internal.gson.GreeStatusResponsePackDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+
+/**
+ * The GreeDevice object repesents a Gree Airconditioner and provides
+ * device specific attributes as well a the functionality for the Air Conditioner
+ *
+ * @author John Cunha - Initial contribution
+ * @author Markus Michels - Refactoring, adapted to OH 2.5x
+ */
+@NonNullByDefault
+public class GreeAirDevice {
+    private final Logger logger = LoggerFactory.getLogger(GreeAirDevice.class);
+    private final static Charset UTF8_CHARSET = Charset.forName("UTF-8");
+    private final static HashMap<String, HashMap<String, Integer>> tempRanges = createTempRangeMap();
+    private boolean isBound = false;
+    private InetAddress ipAddress = InetAddress.getLoopbackAddress();
+    private int port = 0;
+    private String encKey = "";
+    private Optional<GreeScanResponseDTO> mScanResponseGson = Optional.empty();
```

Comment:
Please use camelcase for names.
```suggestion
    private Optional<GreeScanResponseDTO> scanResponseGson = Optional.empty();
```

Reply:
changed